src/native/external/zlib-ng/deflate_fast.c

```diff
@@ -58,7 +58,9 @@ Z_INTERNAL block_state deflate_fast(deflate_state *s, int flush) {
         }
 
         if (match_len >= WANT_MIN_MATCH) {
-            check_match(s, s->strstart, s->match_start, match_len);
+            Assert(s->strstart <= _UI16_MAX,"strstart should fit in uint16_t");
+            Assert(s->match_start <= _UI16_MAX,"match_start should fit in uint16_t");
+            check_match(s, (Pos)s->strstart, (Pos)s->match_start, match_len);
 
```

Comment:
In the deflate*.c files, the `Assert` method is defined only in Debug. This is fine to use because `check_match` is also only defined in Debug.